dcpy/test/lifecycle/ingest/test_extract.py

```diff
@@ -1,36 +1,30 @@
 from pathlib import Path
+from pydantic import TypeAdapter
 import yaml
 
-from dcpy.models import file
+from dcpy.models.lifecycle.ingest import Source, LocalFileSource
 from dcpy.utils import s3
 from dcpy.connectors.edm import publishing
-from dcpy.lifecycle.ingest import (
-    configure,
-    extract,
-)
+from dcpy.lifecycle.ingest import extract, configure
+
 from . import RESOURCES
 
 
 def test_download_file(create_buckets, create_temp_filesystem: Path):
+    """Tests both configure.get_filename and extract.download_file_from_source"""
     with open(RESOURCES / "sources.yml") as f:
-        sources = yaml.safe_load(f)
+        sources = TypeAdapter(list[Source]).validate_python(yaml.safe_load(f))
```

Comment:
@sf-dcp a useful function - this is how I read in https://github.com/NYCPlanning/data-engineering/blob/main/dcpy/test/lifecycle/ingest/resources/sources.yml directly as a list of `ingest.Source` objects